src/mono/mono/metadata/marshal.c

```diff
@@ -3940,10 +3958,54 @@ emit_managed_wrapper_noilgen (MonoMethodBuilder *mb, MonoMethodSignature *invoke
 }
 #endif
 
+static gboolean
+type_is_blittable (MonoType *type)
```

Comment:
There is already a class->blittable flag.

Reply:
We discussed this, but it's not the same because of byref types.